src/afl-fuzz.c

```diff
@@ -2493,10 +2493,11 @@ int main(int argc, char **argv_orig, char **envp) {
 
       for (entry = 0; entry < afl->queued_items; ++entry)
         if (!afl->queue_buf[entry]->disabled)
-          if (afl->queue_buf[entry]->exec_us > max_ms)
-            max_ms = afl->queue_buf[entry]->exec_us;
-
-      afl->fsrv.exec_tmout = max_ms;
+          if ((afl->queue_buf[entry]->exec_us/1000) > max_ms)
+            max_ms = afl->queue_buf[entry]->exec_us/1000;
+      
+      if (max_ms > afl->fsrv.exec_tmout)
```

Comment:
the if needs to be removed. we always want the timeout to be set based on max_ms